lib/puma/thread_pool.rb

```diff
@@ -116,6 +117,10 @@ def spawn_thread
               end
 
               @waiting += 1
+              busy_threads = @spawned - @waiting
+              if busy_threads == 0
```

Comment:
Let's extract these 4 lines into a private method, called `signal_threadpool_empty`.

Reply:
Done in https://github.com/puma/puma/commit/64c98c33796253a52bb5a600bf2e4ac15fcd587d